boat_nav/src/layline_action.py

```diff
@@ -111,28 +88,25 @@ def layline_callback(self, goal):
 		d_perp = abs(a*pos.x + b*pos.y + c)/math.sqrt(a*a+b*b)
 		d_par = math.sqrt(dx*dx+dy*dy-d_perp*d_perp)
 		
-		if self.gtAngle(goal.alt_tack_angle, self.wind_coming):
+		if angles.is_on_left(goal.alt_tack_angle, self.wind_coming):
 			tacking_direction = 1
 		else:
 			tacking_direction = -1
 		self._feedback.status = " Tacking away from mark to hit layline. "
 		rospy.loginfo(rospy.get_caller_id() + self._feedback.status)
-		new_target = self.wind_coming - tacking_direction * self.layline
-		print new_target, self.wind_coming, tacking_direction
-		if new_target < 0:
-			new_target += 360
-		new_target = new_target % 360
+		new_target = angles.normalize(self.wind_coming - tacking_direction * self.layline)
 		self.target_heading = new_target
 		self.target_pub.publish(Float32(self.target_heading))
 		
 		tacking_goal = TackingGoal(direction = tacking_direction)
 		self.tacking_client.send_goal(tacking_goal)
-		
+
 		endtime = rospy.Time.now() + rospy.Duration(10)
 		
+		# TODO: Sometimes this loop abruptly exits for no reason
 		while (self.tacking_client.get_state() is GoalStatus.ACTIVE or\
 			   self.tacking_client.get_state() is GoalStatus.PENDING) and rospy.Time.now() < endtime and not did_hit_midpoint:
-			
+
 			pos = self.cur_pos
 			cur_d_perp = abs(a*pos.x + b*pos.y + c)/math.sqrt(a*a+b*b)
 			if cur_d_perp < d_perp*0.48:
```

Comment:
Magic number 0.48

Reply:
Either add a comment or add a constants section to the utilities class

Reply:
Will add comment with TODO to better tune this. Currently, it's simply "something a lil bit less than a half". If it turns into a tuneable parameter down the road, it should be moved into the parameters.yaml file with the rest of the boat parameters.